### PR TITLE
MUMUP-2580 Add widget from mascot announcement

### PR DESCRIPTION
**Pull request by Doug-Reed**
This functionality adds a widget to the homepage by clicking an action button in a feature announcement.

----

Contributor License Agreement adherence:

- [ X] This Contribution is under the terms of Individual [Contributor License Agreements][] (and also Corporate Contributor License Agreements to the extent applicable) appearing in the [Apereo CLA roster][].

[Apereo CLA roster]: http://licensing.apereo.org/completed-clas
[Contributor License Agreements]: https://www.apereo.org/licensing/agreements

### CODE REVIEW

File: uw-frame-components/portal/messages/controllers.js [hunk 1]
```diff
@@ -575,7 +576,9 @@ define(['angular'], function(angular) {
           if (actionType == addToHome) {
             var slash = url.lastIndexOf('/') + 1;
             var fName = url.substr(slash);
-            PortalAddToHomeService.addToHome(fName);
+            $rootScope.addPortletToHome(fName);
+            actionButton.label = 'Added To Home';
```
Comment (apetro) [inline uw-frame-components/portal/messages/controllers.js]: Prefer sentence case. A sentence, even.

File: uw-frame-components/portal/messages/partials/announcement-mascot.html [hunk 2]
```diff
@@ -45,30 +45,30 @@
...
+          <span>{{ announcement.titleShort }}</span>
+          <span class="menu-item-description">{{ announcement.descriptionShort }}</span>
+          <div ng-if="announcement.moreInfoButton || announcement.actionButton" class="compact-buttons">
+            <md-button class="md-raised md-accent"
+                       ng-if="announcement.actionButton"
+                       ng-disabled="announcement.actionButton.disabled"
+                       ng-click="vm.takeButtonAction(announcement.actionButton)"
+                       target="_blank"
+                       rel="noopener noreferrer">
+              {{ announcement.actionButton.label }}
```
Comment (apetro) [inline uw-frame-components/portal/messages/partials/announcement-mascot.html]: a11y: insufficient context problem: in isolation this button label provides insufficient context to understand what it does. Add *what* to home? If a screen reader is jumping between clickables, it's gonna read this label, and they won't have enough context to evaluate whether to actuate it.

File: uw-frame-components/portal/messages/filters.js [hunk 3]
```diff
@@ -40,22 +40,34 @@ define(['angular'], function(angular) {
...
                   message.actionButton.disabled = true;
+                  return true;
+                } else {
+                  message.actionButton.label = 'Add To Home';
+                  message.actionButton.disabled = null;
+                  return true;
                 }
-             }
+              })
+              .catch(function(text) {
```
Comment (ChristianMurphy) [inline uw-frame-components/portal/messages/filters.js]: :art: unused param

File: uw-frame-components/portal/messages/filters.js [hunk 4]
```diff
@@ -40,22 +40,34 @@ define(['angular'], function(angular) {
...
+              var fName = url.substr(slash);
+              PortalAddToHomeService.inHome(fName).then(function(result, text) {
+                if (result) {
+                  message.actionButton.label = 'Added To Home';
                   message.actionButton.disabled = true;
+                  return true;
+                } else {
+                  message.actionButton.label = 'Add To Home';
+                  message.actionButton.disabled = null;
+                  return true;
```
Comment (ChristianMurphy) [inline uw-frame-components/portal/messages/filters.js]: :art: factor duplicated `return true;` outside of conditionals

File: uw-frame-components/portal/messages/controllers.js (hunk 1 above)
Comment (Doug-Reed) [reply uw-frame-components/portal/messages/controllers.js]: The widget exists
You got this already
Desire leads to suffering
Why you wanna add a thing what you currently posses

-What should the sentence say?

**Pull request merged**